### PR TITLE
fault set changes added

### PR DESCRIPTION
**Pull request by AnikaAgiwal2711**
# Description
This PR contains adding new functionalities for fault set like : creating , modifying, deleting and reading a fault set

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

Created a test file and added the scenarios there.
image

Unit test 
image

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: fault_set.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+	err := pd.client.getJSONWithRetry(
+		http.MethodPost, path, fs, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// ModifyFaultSetName will modify the name of the fault set
```
Comment (shenda1) [inline fault_set.go:63]: ModifyFaultSetPerFrofile

File: fault_set.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	err := pd.client.getJSONWithRetry(
+		http.MethodPost, path, pp, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// ModifyFaultSetName will modify the name of the fault set
+func (pd *ProtectionDomain) ReadFaultSet(id string) (*types.FaultSet, error) {
```
Comment (shenda1) [inline fault_set.go:78]: Please change the comment as per the method name

Comment (AnikaAgiwal2711) [reply fault_set.go]: Done

File: fault_set.go (hunk 1 above)
Comment (AnikaAgiwal2711) [reply fault_set.go]: Done

File: fault_set.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
+// Copyright © 2019 - 2022 Dell Inc. or its subsidiaries. All Rights Reserved.
```
Comment (adarsh-dell) [inline fault_set.go:1]: If it is a new file that we are introducing in 2023 then 
// Copyright © 2019 - 2022 Dell Inc. => // Copyright © 2023 Dell Inc.
What do you think about it?
This is applicable for all files. If it was created in 2019 and we are modifying it in 2023 then // Copyright © 2019 - 2023.

Comment (AnikaAgiwal2711) [reply fault_set.go:1]: Makes sense. Thanks

Comment (AnikaAgiwal2711) [reply fault_set.go]: Addressed. Please check

**Review by VamsiSiddu-7 — CHANGES_REQUESTED** [review]
Can you please add unit tests

**Comment by AnikaAgiwal2711** [comment]
> Changes looks good to me. Can you please add unit tests

Sure Vamsi. Let me add

**Comment by AnikaAgiwal2711** [comment]
added the unit test

**Review by shenda1 — APPROVED** [review]

**Review by VamsiSiddu-7 — APPROVED** [review]

**Pull request merged**